_posts/2019-07-30-censored-data/censored_data_tfp.Rmd

```diff
@@ -0,0 +1,724 @@
+# lm predictions on censored data
+pred_c_lm <- predict(fit, df_c %>% select(- intercept))
+```
+
+For the purpose of a simple sanity check, we just get point predictions from our model:
+
+```{r}
+# model predictions on non-censored data, making use of posterior mean
+pred_nc_mcmc <- df_nc %>% as.matrix() %*% summary$mean
+# model predictions on censored data, making use of posterior mean
+pred_c_mcmc <- df_c %>% as.matrix() %*% summary$mean
+```
+
+Let's see predictions next to each other, starting with the non-censored data:
+
+```{r}
+comp_nc <- tibble(
+  actual = as.vector(check_time_nc),
+  predicted_lm = as.vector(pred_nc_lm),
+  predicted_mcmc = as.vector(pred_nc_mcmc)
+)
+head(comp_nc, 10)
+```
+
+```
+# A tibble: 10 x 3
+   actual predicted_lm predicted_mcmc
+    <dbl>        <dbl>          <dbl>
+ 1     49         85.5           78.5
+ 2    101        136.           142. 
+ 3    292        162.           154. 
+ 4     21         58.1           49.8
+ 5    103         91.9           88.9
+ 6     46         73.7           69.7
+ 7     78        106.            96.6
+ 8     91         82.3           79.9
+ 9     47         60.0           52.7
+10    196        122.           126. 
+ 
+```
+
+It seems fair to say that for both models, predictions can be pretty far off, but it is interesting -- and reassuring -- to see how similar they are. We could also compare MSEs:
+
+```{r}
+mse_lm <- ((comp_nc$predicted_lm - comp_nc$actual)^2) %>% mean()
+mse_mcmc <- ((comp_nc$predicted_mcmc - comp_nc$actual)^2) %>% mean()
+c(mse_lm, mse_mcmc)
+```
+
+```
+[1] 6273.799 6477.195
+```
+
+Doing the same for the censored data:
+
+```{r}
+comp_c <- tibble(
+  actual = as.vector(check_time_c),
+  predicted_lm = as.vector(pred_c_lm),
+  predicted_mcmc = as.vector(pred_c)
+)
+head
+```
+
+
+```
+# A tibble: 10 x 3
+   actual predicted_lm predicted_mcmc
+    <dbl>        <dbl>          <dbl>
+ 1     10         91.4           91.1
+ 2     80         87.8           82.3
+ 3     38         61.0           53.9
+ 4    304        673.           893. 
+ 5     89        105.           104. 
+ 6     68         89.7           92.8
+ 7      5         71.6           64.3
+ 8     77         68.0           61.6
+ 9     89        125.           122. 
+10    174        119.           117. 
+ 
+```
+
+```{r}
+mse_lm <- ((comp_c$predicted_lm - comp_c$actual) ^ 2) %>% mean()
+mse_mcmc <- ((comp_c$predicted_mcmc - comp_c$actual) ^ 2) %>% mean()
+c(mse_lm, mse_mcmc)
+```
+
+```
+
+[1]  9273.062 11328.088
+```
+
+Overall, we conclude that the model works reasonably well.
+
+## Wrapup
+
+We've shown how to model censored data -- resp. a frequent subtype thereof involving durations -- using `tfprobability`. The `check_times` data from `parsnip` were a fun choice, but this modeling technique may be even more useful when censoring is more substantial. Hopefully his post has provided some guidance on how to handle censored data in your own work. Thanks for reading!
```

Comment:
What does `resp.` stand for here?

Reply:
respectively - I wasn't sure either if this was okay ... I was looking for an alternative to "or" ... would you have an idea?

Reply:
I could change it to "or rather, a frequent subtype..."?
Would that be better?

Reply:
Yes!

Reply:
Okay thanks!